### PR TITLE
Feature lwdev 8703 send positions reports on daily basis

### PR DESCRIPTION
**Pull request by martin8282**

### CODE REVIEW

File: src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs [hunk 1]
```diff
@@ -142,8 +142,149 @@ public async Task PerformReporting()
...
             IReadOnlyDictionary<string, string> emails)
+        {
+            var failedNotifications = new Dictionary<DailyTradingNotification, Exception>();
+            Exception anyException = null;
+            foreach (var notification in notifications)
+            {
+                try
+                {
+                    await _emailService.PrepareAndSendEmailAsync(emails[notification.ClientId],
+                        $"Margin Trading - Dayly trading report for {notification.CurrentDay}",
```
Comment (GSailen) [inline src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs]: Daily ;)

File: src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs [hunk 2]
```diff
@@ -142,8 +142,149 @@ public async Task PerformReporting()
...
+                    .Where(x => x.ClientId == clientId && accountIds.Contains(x.AccountId))
+                    .OrderByDescending(x => x.OpenDate).ToList(),
+                PendingPositions = pendingPositions
+                    .Where(x => x.ClientId == clientId && accountIds.Contains(x.AccountId))
+                    .OrderByDescending(x => x.CreateDate).ToList(),
+                Accounts = filteredAccounts,
+            };
+        }
+
+        private async Task SendNotificationsMonthly(IEnumerable<MonthlyTradingNotification> notifications,
```
Comment (GSailen) [inline src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs]: SendNotifications and SendNotificationsMonthly logic is quite the same, except time period labels. Maybe they can be merged together?
PerformReporting and PerformReportingMonthly are quite the same too. Let's extract the same part into private method and reuse it.

**Review by GSailen — CHANGES_REQUESTED** [review]

File: src/MarginTrading.NotificationGenerator/Startup.cs [hunk 3]
```diff
@@ -136,10 +136,13 @@ private async Task StartApplication()
                 var registry = new Registry();
                 var settingsCalcTime = ApplicationContainer.Resolve<NotificationGeneratorSettings>()
                     .MonthlyTradingReportSettings.InvocationTime;
-                
-                //TODO ToRunEvery(0) FOR TEST PURPOSES
+
                 registry.Schedule<MonthlyTradingReportJob>().ToRunEvery(1).Months().On(1)
                     .At(settingsCalcTime.Hours, settingsCalcTime.Minutes);
+
```
Comment (GSailen) [inline src/MarginTrading.NotificationGenerator/Startup.cs]: It seems that settings should be separated for Daily/Monthly reports.

**Review by GSailen — CHANGES_REQUESTED** [review]

File: src/MarginTrading.NotificationGenerator/Startup.cs (hunk 3 above)
Comment (martin8282) [reply src/MarginTrading.NotificationGenerator/Startup.cs]: addressed

File: src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs (hunk 2 above)
Comment (martin8282) [reply src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs]: yup, refactored as recommended and possible

Comment (GSailen) [reply src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs]: Ok, thats fine then.

File: src/MarginTrading.NotificationGenerator/Startup.cs [hunk 4]
```diff
@@ -137,11 +137,14 @@ private async Task StartApplication()
                 var settingsCalcTime = ApplicationContainer.Resolve<NotificationGeneratorSettings>()
                     .MonthlyTradingReportSettings.InvocationTime;
 
+                var settingsDailyCalcTime = ApplicationContainer.Resolve<NotificationGeneratorSettings>()
+                    .MonthlyTradingReportSettings.InvocationTime;
```
Comment (GSailen) [inline src/MarginTrading.NotificationGenerator/Startup.cs]: That's for daily... not monthly

File: src/MarginTrading.NotificationGenerator/Modules/JobModule.cs [hunk 5]
```diff
@@ -52,7 +52,9 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterInstance(_settings).As<NotificationGeneratorSettings>().SingleInstance();
             
             builder.RegisterInstance(_settings.MonthlyTradingReportSettings.Filter).AsSelf().SingleInstance();
-            
+
+            builder.RegisterInstance(_settings.DailyTradingReportSettings.Filter).AsSelf().SingleInstance();
```
Comment (GSailen) [inline src/MarginTrading.NotificationGenerator/Modules/JobModule.cs]: It will never resolve correctly

**Review by GSailen — CHANGES_REQUESTED** [review]

File: src/MarginTrading.NotificationGenerator/Modules/JobModule.cs (hunk 5 above)
Comment (martin8282) [reply src/MarginTrading.NotificationGenerator/Modules/JobModule.cs]: addressed

**Review by andrey-anokhin — APPROVED** [review]

**Review by GSailen — APPROVED** [review]

**Pull request merged**